OneCommunityGlobal/HGNRest
# Luis fix user profile projects delay

# Description

<img width="1470" alt="Screenshot 2023-11-05 at 7 42 52 PM" src="https://github.com/OneCommunityGlobal/HGNRest/assets/22454953/76798e5d-42fa-436e-8534-a02f5d2711d9">


## Related PRS (if any):
To test this backend PR you need to checkout development branch 
…

## Main changes explained:
On the backend I added a check to remove cache.
The issue was the backend was sending back old cache instead of clearing it when a project was added.
Hence it was sending old data.
The change occurs in ProjectController.js inside of the assignProjectToUsers() when mapping over each element cache is checked if so it removes it and adds the user.


## How to test:

1. check into current branch
2. The frontend should be development, check into and run`npm install`
3. **Note** if you are getting the WBS error, you may need to pull the latest development branch
4. Run on the backend `npm run install`,  then build and run the server `npm run build && npm run start`
5. log as admin user or owner account
6. go to view-profile→ projects→ take note how many projects are there.
7. Next go to other links → projects → select a project → add your current user that you are logged in as.
8. Add the user then navigate to the view-profile→ projects→ and check if the project was added to the list of projects.
9. Next click timelog check to see if the project is in the list of tasks.
10. click "add tangible time entry"
11. click on projects/task and see if the newly added project is inside of the drop down menu

## Screenshots or videos of changes:
https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/22454953/96cb30a8-34a6-43d9-89a1-39998f196a21


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/22454953/59580642-563d-44a8-9a70-9091121d4c65


Updated video

https://github.com/OneCommunityGlobal/HGNRest/assets/22454953/b1e9172a-d0bb-442a-831d-29c9ba7784ea




## Note:
There was a front end branch but for some reason, it removed the projects. 
Upon testing the development branch worked just fine. So, I may have adjusted the frontend which caused the issue. 
However, checking ino the development branch then pull this backend, fixes everything. 
There are a few errors that pop up on the frontend please disregard them as they are irrelevant to this PR.
<img width="344" alt="Screenshot 2024-04-11 at 7 48 51 PM" src="https://github.com/OneCommunityGlobal/HGNRest/assets/22454953/557005a6-1a73-4224-9e68-fd0737eaf2ee">
